bundles/org.openhab.binding.monopriceaudio/src/main/java/org/openhab/binding/monopriceaudio/internal/communication/MonopriceAudioZone.java

```diff
@@ -48,10 +49,23 @@
 
     private String zoneId;
 
-    // make a list of all valid zone ids
-    public static final List<String> VALID_ZONES = Arrays.stream(values()).filter(z -> z != ALL).map(z -> z.getZoneId())
+    // make a list of all valid zone names
+    public static final List<String> VALID_ZONES = Arrays.stream(values()).filter(z -> z != ALL).map(z -> z.name())
             .collect(Collectors.toList());
 
+    // make a list of all valid zone ids
+    public static final List<String> VALID_ZONE_IDS = Arrays.stream(values()).filter(z -> z != ALL)
+            .map(z -> z.getZoneId()).collect(Collectors.toList());
+
+    public static MonopriceAudioZone fromZoneId(String zoneId) throws MonopriceAudioException {
+        for (MonopriceAudioZone z : MonopriceAudioZone.values()) {
+            if (z.zoneId.equalsIgnoreCase(zoneId)) {
+                return z;
+            }
+        }
+        throw new MonopriceAudioException("Invalid zoneId specified: " + zoneId);
```

Comment:
The code begs for streams :) If you like, try findAny().orElseThrow(...

Reply:
got it!